### DISCUSSION TITLE
VSCodium

### DISCUSSION THREAD

**Discussion by chipset2011** [zepp-health/zeppos-forum#38] [Q&A]
hello, is there someone who can help me with setting up vs code? I have a number of questions that could not be solved by searching the Internet, who can I contact to discuss the details?

**Comment by htoooth** [comment]
@mrcoon please help this developer.

**Comment by mrcoon** [comment]
@chipset2011 please tell us more about the problems you are experiencing.

**Reply by chipset2011** [reply]
@mrcoon  i tried to try to run hello world on gtr 4, created a project for gtr3, got a qr code, but after scanning in the application I get the error "download failed, error code: null value". And I didn't understand how to load the compiled application into the emulator using the same VSCode.

**Reply by mrcoon** [reply]
please submit feedback on the Zepp APP and tell us your user ID so that we can find the problem faster.

**Reply by chipset2011** [reply]
ID: 7002644776

**Reply by mrcoon** [reply]
@chipset2011 Sorry, we did not retrieve your feedback by the user ID you provided, please check if it is correct.

**Reply by chipset2011** [reply]
mrcoon It would probably be easier if you made a video on how to properly set up and run hello world through vs Codium, but probably it would be impudence on my part.  If you don't mind making a video, that would be great.

**Reply by chipset2011** [reply]
mrcoon I have created a report, please take a look

**Reply by mrcoon** [reply]
@chipset2011 I just checked the feedback you submitted yesterday, but it doesn't seem to have been submitted to the log as I only see a screenshot of the error message.
From the information you describe, we suspect that the device you are using and the previewed package do not match, causing the download to fail. This is because you created a GTR3 project but need to install it to a GTR4 watch. Please check if the `app.json` contains the `deviceSource` for GTR4.
Device base information document: https://docs.zepp.com/docs/reference/related-resources/device-list/

**Reply by mrcoon** [reply]
> mrcoon It would probably be easier if you made a video on how to properly set up and run hello world through vs Codium, but probably it would be impudence on my part. If you don't mind making a video, that would be great.

We have provided some videos, please refer to: [Zepp OS TV](https://developer.zepp.com/os/zepp-os-tv).
Hope this will help you
